StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -616,8 +641,57 @@ extension PaymentSheet {
             } else {
                 fatalError("Either confirmHandler or confirmHandlerForServerSideConfirmation must be non-nil when using deferred intents")
             }
+        }
+    }
+
+    private static func confirmClientSide(for deferredContext: DeferredIntentContext, with clientSecret: String) {
+        // Retrieve either the payment or setup intent, then confirm it client side
+        deferredContext.configuration.apiClient.retrieveIntent(for: deferredContext.intentConfig,
+                                                                  withClientSecret: clientSecret) { result in
```

Comment:
We can follow up in another PR, but we should try to avoid this retrieval.

Looks like besides the client_secret we inspect the PI `shipping`, but I don't think we need to do that for the deferred case.

Reply:
This implies we can confirm an intent w/o actually having the intent object...?

Reply:
My understanding was that we split these into two handlers so we could avoid this retrieval for SSC but it was needed for CSC.

Reply:
> This implies we can confirm an intent w/o actually having the intent object...?
Right. You won't find the object being passed into the `confirmPaymentIntent` call, just the client secret inside the `STPPaymentIntentParams` object.

> My understanding was that we split these into two handlers so we could avoid this retrieval for SSC but it was needed for CSC.
It's the other way around.  If you look at `STPPaymentHandler.handleNextAction`, the first thing it does is retrieve the Intent to inspect the status.  But per ^, no need to do this for CSC.

Reply:
I introduced `Intent.Confirmable` which is a minimum subset of a `Intent` to confirm an intent.